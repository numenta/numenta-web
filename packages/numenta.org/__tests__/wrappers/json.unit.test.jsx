import React from 'react'
import renderer from 'react-test-renderer'
import stubContext from 'react-stub-context'

import context from '__tests__/__mocks__/reactContextMock'

import Json from '../../wrappers/json'

const route = {
  page: {
    data: {
      title: 'Wrapper Title',
      body: 'Wrapper Body',
    },
    file: {},
    path: '',
  },
}


describe('JSON Wrapper React component', () => {

  it('Renders correctly', () => {
    const JsonStubbed = stubContext(Json, context)
    const component = renderer.create(
      <JsonStubbed route={route}>
        ["hello": "world"]
      </JsonStubbed>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
