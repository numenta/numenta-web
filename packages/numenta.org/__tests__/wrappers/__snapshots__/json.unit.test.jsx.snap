exports[`JSON Wrapper React component Renders correctly 1`] = `
<div
  className="json">
  <h1>
    Wrapper Title
  </h1>
  <p>
    Raw view of json file
  </p>
  <pre
    dangerouslySetInnerHTML={
      Object {
        "__html": "{
          \"title\": \"Wrapper Title\",
          \"body\": \"Wrapper Body\"
      }",
      }
    } />
</div>
`;
